src/Symfony/Component/Routing/Loader/AnnotationDirectoryLoader.php

```diff
@@ -76,6 +76,15 @@ function (\SplFileInfo $current) {
      */
     public function supports($resource, $type = null)
     {
+        if ($type && 'annotation' !== $type) {
+            return false;
+        }
+
+        // classes are handled by the AnnotationClassLoader
+        if (preg_match('/^(?:\\\\?[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*)+$/', $resource)) {
```

Comment:
wondering why it worked on 3.3 without this? Was the order of the loaders different?

Reply:
Actually, I think we really need to make the change in 3.3. The observed behaviour seems to be caused by #25113 which just was not part of a patch release of 3.3 yet.

Reply:
Ah ok I see :wink: 

Can't we do a `class_exists` call to see if `$resource` is a class instead?